foqus_lib/framework/graph/node.py

```diff
@@ -143,10 +143,32 @@ def torch_float(*args, **kwargs):
     return torch_load, torch_tensor, torch_float
 
 
+def attempt_load_pickle(try_imports=True):
```

Comment:
I've just double-checked to be sure, but `pickle` is part of the Python standard library, so I don't think we need to check and/or handle the case when it might not be installed (this applies here, as well elsewhere in the code).

For this reason, I'd recommend to remove the import-checking code before merging. However, since strictly speaking they're only unnecessary rather than incorrect, we could also consider keeping them in case their presence makes other parts of the code significantly simpler.

Reply:
I will change this to check for `sklearn` instead, since this is the only context in which `pickle_load()` is called, and update the test skip markers appropriately.